### ISSUE TITLE
Transdroid 'forgets' search is installed

### ISSUE THREAD

**Issue by cathelijne** [erickok/transdroid#417]
Hoi,

Whenever I perform a search in Transdroid (2.5.8), it prompts me to install search. But search is installed.
I can remove search from my system (Oneplus 3, OxygenOS 5.0.0, Android 8.0.0, SuperSU rooted), and download it and have it installed by Transdroid when it asks for search to be installed. This will work, until I exit the app. Quite a cumbersome way to make it work. When I load Transdroid after, search is 'gone'.

Not really into Android development much, but I know my way around the command line and have platform tools installed. Let me know what else you need to know.

**Comment by erickok** [comment]
That's really weird, I have never seen that before. Transdroid just tries to perform a query to see if the module is installed. Could you check into Setting -> Help -> Send error report and see if your log includes a hint?

**Comment by erickok** [comment]
I have not yet seen a log from you , but you can check if you see lines such as 
```
Failed to find provider info for org.transdroid.search.torrentsitesprovider
```
in your error log. If so, Transdroid cannot find any installation of the Search Module.

**Comment by cathelijne** [comment]
Aaargh, terribly sorry. Got sidetracked with the flu and hadn't gotten around to it.
My error log holds all normal startup entries, like what's the hostname of the ip-address, are there any torrents running, etc. 

`7289 -- Mon Feb 19 13:56:01 GMT+01:00 2018 -- 3 -- Skip the RSS checker service, as background data is disabled, the service is disabled or we are not connected.
7290 -- Mon Feb 19 13:56:01 GMT+01:00 2018 -- 3 -- Success: {"id": 2, "result": {"stats": {"upload_protocol_rate": 726, "max_upload": 80.0, "download_protocol_rate": 366, "download_rate": 0, "has_incoming_connections": false, "num_connections": 0, "max_download": -1.0, "upload_rate": 0, "dht_nodes": 167, "free_space": 198287822848, "max_num_connections": -1}... (608 chars)
7291 -- Mon Feb 19 13:56:01 GMT+01:00 2018 -- 3 -- home: Retrieved torrent listing
7292 -- Mon Feb 19 13:56:01 GMT+01:00 2018 -- 3 -- home: 0 new torrents, 0 newly finished torrents.`

Looks all fairly standard to me. The only one relating to search comes immediately after and reads like this:

`7293 -- Mon Feb 19 13:56:02 GMT+01:00 2018 -- 3 -- Cannot retrieve latest app or search module version code from the site: java.lang.NumberFormatException: For input string: "404: Not Found"`

**Comment by Atrejoe** [comment]
The same happens for me (this was already happening in the previous version). There are no errors mentions in the debug log, just regular traffic an tracing messages, that appear to be unrelated. (the word provider does not appear anywhere)

Unless I manually start up Transdroid search first, Transdroid does not known the search module in installed.

**Comment by Atrejoe** [comment]
Yes I am, Android 8.

If you have a version producing more more tracing information, I'm 
willing to install it.

**Comment by cathelijne** [comment]
@Atrejoe Your comment that you have to start search manually gets me thinking... Since Android 8 I have to disable battery optimization for some apps to keep their widgets working etc. If I _disable battery optimization for Transdroid Search_, search from the trandroid app works.

I just rebooted my phone to try and make sure the search app wasn't still running and it works.

**Comment by erickok** [comment]
Yes it's what I though. The ContentProvider from Search is being found when looking for it:
https://github.com/erickok/transdroid/blob/master/app/src/main/java/org/transdroid/core/app/search/SearchHelper.java#L76

I wonder it'd be enough to just try a query instead. I mean, something must kick it out of the doze mode, right? 🤞

**Comment by erickok** [comment]
Note to self: https://developer.android.com/training/monitoring-device-state/doze-standby.html#testing_doze_and_app_standby

**Comment by cathelijne** [comment]
Well querying from the app in its current form doesn't wake it up. But opening the settings (helper?) app does. So something should be possible :)

Anyway, for now, a good workaround is to open the search settings, or disable battery optimization. Both work for me. :)

**Comment by Atrejoe** [comment]
Disabling battery optimization for Transdroid search works for me too.

**Comment by kwaazaar** [comment]
Running into this issue now.

I'm not sure, but RARBG search hardly ever works (I retry about 10 times to get results) and it was my default search engine. 
Configuring another search engine is also broken, but directly after reinstalling the search module it worked. I switched to another search engine and after that the search worked again. Also after switching to RARBG again. Let's see how long it keeps working.
